src/Symfony/Component/DependencyInjection/Tests/Fixtures/config/prototype.expected.yml

```diff
@@ -10,7 +10,8 @@ services:
         tags:
             - { name: foo }
             - { name: baz }
-        deprecated: '%service_id%'
+        deprecated:
+            message: '%service_id%'
```

Comment:
I think we should mandate all arguments (ie trigger a deprecation when a string is used as allowed before, and an exception when not all arguments are passed when an array is used.
What about omitting the keys?
`deprecation: ['vendor/foo', '1.2', 'message', 'id', etc]`

Reply:
Is it OK to omit keys only for YAML? in XML we still have attributes
What about readability (or auto-completion for IDEs) ?